controller/playground/manager.py

```diff
@@ -0,0 +1,349 @@
+            futures = [
+                executor.submit(
+                    __get_most_similar_records,
+                    project_id,
+                    embedding_id,
+                    questions_tensors[index],
+                    EVALUATION_RUN_LIMIT_DEFAULT,
+                    None,
+                    threshold,
+                )
+                for index, evaluation_set in enumerate(evaluation_sets)
+            ]
+
+            search_results = [future.result() for future in futures]
+
+        all_record_ids = set()
+        for evaluation_set, search_result in zip(evaluation_sets, search_results):
+            expected_record_ids = {str(rid) for rid in evaluation_set.record_ids}
+            received_record_ids = {str(record["id"]) for record in search_result}
+            all_record_ids.update(expected_record_ids | received_record_ids)
+
+        all_records = record_db_bo.get_by_record_ids(project_id, list(all_record_ids))
+        record_map = {
+            str(record.id): sql_alchemy_to_dict(record) for record in all_records
+        }
+
+        for evaluation_set, search_result in zip(evaluation_sets, search_results):
+            expected_record_ids = {str(rid) for rid in evaluation_set.record_ids}
+            received_record_ids = {str(record["id"]) for record in search_result}
+
+            true_positives = expected_record_ids & received_record_ids
+            false_negatives = expected_record_ids - received_record_ids
+            false_positives = received_record_ids - expected_record_ids
+
+            result = {
+                "evaluation_set_id": str(evaluation_set.id),
+                "true_positives": to_frontend_obj_raw(
+                    [record_map[record_id] for record_id in true_positives]
+                ),
+                "false_positives": to_frontend_obj_raw(
+                    [record_map[record_id] for record_id in false_positives]
+                ),
+                "false_negatives": to_frontend_obj_raw(
+                    [record_map[record_id] for record_id in false_negatives]
+                ),
+            }
+            evaluation_results.append(result)
+        state = EVALUATION_RUN_STATE.SUCCESS
+    except Exception:
+        state = EVALUATION_RUN_STATE.FAILED
+    evaluation_run_db_bo.update(
+        project_id, evaluation_run.id, state, evaluation_results, None, True
+    )
+    return evaluation_run
+
+
+def __get_tensors_for_texts(
+    refinery_project_id: str, embedding_id: str, texts: List[str]
+) -> Tuple[bool, Optional[List[Any]]]:
+
+    url = f"{EMBEDDING_SERVICE}/calc-tensor-by-pkl/{refinery_project_id}/{embedding_id}"
+
+    response = requests.post(
+        url,
+        headers={
+            "accept": "application/json",
+            "content-type": "application/json",
+        },
+        json={"texts": texts},
+    )
+    if response.ok:
+        return response.json()["tensor"]
+    else:
+        return None
```

Comment:
usually we use service_requests.post_call_or_raise in a new function inthe controller>embedding>connector.py for requests between our containers